tests/mocha/contextmenu_items_test.js

```diff
@@ -0,0 +1,430 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+suite('Context Menu Items', function() {
+  var workspace;
+  var registry;
```

Comment:
I went ahead and changed to the property style (this.registry etc). But for the record the reasons I do this:
a) It's shorter. I just added `this` over 150 times to this file.
b) It makes more sense for variable scope. There's no reason that e.g. undoOption needs to be a property on the whole test file, it makes more intuitive sense (to me, anyway) for it to be a local variable scoped to the undo suite.
c) This is how I've always done it in JsUnit :P (I know that's not a good reason, but it is a reason haha)

It makes sense for things like workspace where you're modifying it outside of this test file and that's an easy way to access the workspace from the test helpers. But it seems like overkill to do that for every single variable.

However I changed it for consistency in style with other tests but I'm not sure why the property style is the default instead of using local vars. Here's a random example of this from the Mocha docs: https://mochajs.org/#asynchronous-hooks

Reply:
I'll defer on Monica to this, but I thought that using a setup function and putting it on `this` within the setup function guarantees that it's reset per suite, and is actually on a fairly small scope (not whole-file).

If that's true, the reason to do it this way is to avoid accidentally sharing state between tests that are in the same suite.

Reply:
It's going to be reset per suite no matter what I think because you'd be defining it in the setup still, just declaring it outside of setup so you can use it in the following functions as well. Here's I think a difference:

Option 1:

```
suite('Unicorn', function() {
  var unicorn;
  setup(function() {
     unicorn = new Unicorn();
  });
  test('Unicorn has a horn', function() {
    chai.assert.isTrue(unicorn.hasHorn());
  });
});

suite('Leprechaun', function() {
   test('Leprechaun exists', function() {
      // This is a bug; your IDE puts red squiggles under unicorn,
      // and when you run it you get "ReferenceError: unicorn is not defined"
      // so it's very easy to tell you're using something you haven't set up yet
     chai.assert.exists(unicorn);
    });
);
```

If you do it the other way with `this.unicorn = new Unicorn();` then later on in the Leprechaun suite, `this.unicorn` will just be undefined. So either way, it does get set up properly only on the correct tests, and it does get reset between tests (because you're declaring it in the setup function), but the behavior where it's not even a declared variable in the wrong scope seems more intuitive to me, personally.

I think in the end it isn't that big of a deal but I was just explaining why that's the approach I've used in the past since Monica asked my reasoning. :)

Reply:
Good point. In that case, I'm down for either, and we don't have to decide during this PR review.

Reply:
Thanks for the explanation. Yes, the scope is limited in the suite (so also if you defined something in test, it is only accessible in teardown of the test suite that the test is defined in).

I'm open to changing to the var instead of this, but I'd rather it something we decide for all the tests as it seems to me mainly a formatting reason. The reason why I prefer `this` at this time is because some variables are defined on `this` (mainly `eventsFireStub` `clock` `workspace` because its more convenient to share the this context to  shared setup rather than passing each one. However I think an argument can be made either way about variables that don't need to be touched by shared setup/teardown (that var would make it easier to differentiate, and this would make less overhead because there's no choice).